node_modules/col-core/lib/storage.js

```diff
@@ -162,21 +160,36 @@ var getObjectMetadata = module.exports.getObjectMetadata = function(key, callbac
 };
 
 /**
- * @param  {String}     uri         URL or S3 Object Key or null
- * @return {Boolean}                True if uri is S3 Object Key and not a proper URL
+ * @param  {String}     uri         S3 address or other
+ * @return {Boolean}                True URI matches s3://my-bucket/my-object-key
  */
-var isS3ObjectKey = module.exports.isS3ObjectKey = function(uri) {
-  return !!uri && !_.startsWith(uri, 'http');
+var isS3Uri = module.exports.isS3Uri = function(uri) {
+  return _.startsWith(uri, 's3://');
 };
 
 /**
- * Get URL of resource
- *
- * @param  {Object}     uri         S3 Object Key or an absolute URL
- * @return {String}                 URL with which you can download the file
+ * @param  {String}     bucket              S3 bucket name
+ * @param  {String}     s3ObjectKey         S3 Object Key
+ * @return {String}                         S3 address (e.g., s3://my-bucket/my-object-key)
  */
-var getUrl = module.exports.getUrl = function(uri) {
-  return isS3ObjectKey(uri) ? config.get('aws.s3.baseDownloadUrl') + '/' + uri : uri;
+var buildS3Uri = module.exports.buildS3Uri = function(bucket, s3ObjectKey) {
+  return 's3://' + bucket + '/' + s3ObjectKey;
+};
+
+/**
+ * @param  {String}     s3Uri          URI of object in S3 (e.g., s3://my-bucket/my-object-key)
+ * @return {String}                    Strictly the S3 Object Key
+ */
+var extractS3ObjectKey = function(s3Uri) {
+  return s3Uri.split('/').slice(3).join('/');
+};
+
+/**
+ * @param  {String}     s3Uri          URI of object in S3 (e.g., s3://my-bucket/my-object-key)
+ * @return {String}                    Strictly the S3 bucket name
+ */
+var extractS3Bucket = function(s3Uri) {
+  return s3Uri.split('/')[2];
 };
```

Comment:
@johncrossman This isn't the biggest deal in the world, but it strikes me that since `extractS3ObjectKey` and `extractS3Bucket` are always called together, perhaps they should just be one function? Then you'd only have to call `split('/')` once and could return both values.